File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s115__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s115_fp64(double *restrict a, double *restrict aa, int64_t LEN_2D, uint8_t *workspace, int64_t workspace_bytes) {
    (void)workspace;
    (void)workspace_bytes;
    const int64_t BS = 128;
    if (LEN_2D <= 256) {
        #pragma omp parallel
        {
            for (int64_t j = 0; j < LEN_2D; ++j) {
                const double aj = a[j];
                #pragma omp for simd schedule(static)
                for (int64_t i = j + 1; i < LEN_2D; ++i) {
                    a[i] -= aa[j * LEN_2D + i] * aj;
                }
            }
        }
        return;
    }
    #pragma omp parallel
    {
        for (int64_t jb = 0; jb < LEN_2D; jb += BS) {
            int64_t je = jb + BS < LEN_2D ? jb + BS : LEN_2D;
            #pragma omp single
            {
                for (int64_t j = jb; j < je; ++j) {
                    const double aj = a[j];
                    for (int64_t i = j + 1; i < je; ++i) {
                        a[i] -= aa[j * LEN_2D + i] * aj;
                    }
                }
            }
            #pragma omp for simd schedule(guided)
            for (int64_t i = je; i < LEN_2D; ++i) {
                for (int64_t j = jb; j < je; ++j) {
                    a[i] -= aa[j * LEN_2D + i] * a[j];
                }
            }
        }
    }
}

File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-fortran-skills/tsvc_2_s115__submitted.f90
subroutine tsvc_2_s115_fp64(a, aa, LEN_2D, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_2D
  real(c_double), intent(inout) :: a(LEN_2D)
  real(c_double), intent(in) :: aa(LEN_2D, LEN_2D)
  type(c_ptr), value, intent(in) :: workspace
  integer(c_int64_t), value, intent(in) :: workspace_size
  integer(c_int64_t) :: i, j
  real(c_double), volatile :: prod
  !$omp parallel private(i, j, prod)
  do j = 1, LEN_2D - 1
    !$omp do simd schedule(static)
    do i = j + 1, LEN_2D
      prod = aa(i, j) * a(j)
        a(i) = a(i) - prod
    end do
    !$omp end do simd
  end do
  !$omp end parallel
end subroutine tsvc_2_s115_fp64
